### ISSUE TITLE
Cannot update RQuantLib

### ISSUE THREAD

**Issue by waynelapierre** [eddelbuettel/rquantlib#188]
I have been using RQuantLib for several years, but this morning it stop being able to be updated. The error message is:

```
  In file included from /usr/include/ql/experimental/inflation/piecewiseyoyoptionletvolatility.hpp:30,
                 from /usr/include/ql/experimental/inflation/interpolatedyoyoptionletstripper.hpp:28,
                 from /usr/include/ql/experimental/inflation/all.hpp:7,
                 from /usr/include/ql/experimental/all.hpp:17,
                 from /usr/include/ql/quantlib.hpp:46,
                 from ../inst/include/rquantlib_internal.h:23,
                 from curves.cpp:24:
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::LogLinear, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::LogLinear const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:260:66:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM[(struct IterativeBootstrap &)&D.1295937].n_’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:260:66: note: ‘<anonymous>’ declared here
  260 |                                     std::vector<QuantLib::Date>()));
      |                                                                  ^
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::LogLinear, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::LogLinear const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:260:66:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM <vector(2) long unsigned int> [(long unsigned int *)&D.1295937 + 248B]’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:260:66: note: ‘<anonymous>’ declared here
  260 |                                     std::vector<QuantLib::Date>()));
      |                                                                  ^
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::Cubic, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::Cubic, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::Cubic, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::Cubic const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::Cubic, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:267:62:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM[(struct IterativeBootstrap &)&D.1297654].n_’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:267:62: note: ‘<anonymous>’ declared here
  267 |                                 std::vector<QuantLib::Date>()));
      |                                                              ^
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::Cubic, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::Cubic, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::Cubic, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::Cubic const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ForwardRate, QuantLib::Cubic, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:267:62:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM <vector(2) long unsigned int> [(long unsigned int *)&D.1297654 + 248B]’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:267:62: note: ‘<anonymous>’ declared here
  267 |                                 std::vector<QuantLib::Date>()));
      |                                                              ^
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Linear, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Linear, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Linear, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::Linear const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Linear, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:274:63:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM[(struct IterativeBootstrap &)&D.1297794].n_’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:274:63: note: ‘<anonymous>’ declared here
  274 |                                  std::vector<QuantLib::Date>()));
      |                                                               ^
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Linear, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Linear, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Linear, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::Linear const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Linear, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:274:63:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM <vector(2) long unsigned int> [(long unsigned int *)&D.1297794 + 248B]’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:274:63: note: ‘<anonymous>’ declared here
  274 |                                  std::vector<QuantLib::Date>()));
      |                                                               ^
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::LogLinear, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::LogLinear const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:281:66:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM[(struct IterativeBootstrap &)&D.1299528].n_’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:281:66: note: ‘<anonymous>’ declared here
  281 |                                     std::vector<QuantLib::Date>()));
      |                                                                  ^
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::LogLinear, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::LogLinear const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::LogLinear, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:281:66:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM <vector(2) long unsigned int> [(long unsigned int *)&D.1299528 + 248B]’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:281:66: note: ‘<anonymous>’ declared here
  281 |                                     std::vector<QuantLib::Date>()));
      |                                                                  ^
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Cubic, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Cubic, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Cubic, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::Cubic const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Cubic, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:288:62:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM[(struct IterativeBootstrap &)&D.1301262].n_’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:288:62: note: ‘<anonymous>’ declared here
  288 |                                 std::vector<QuantLib::Date>()));
      |                                                              ^
In member function ‘QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Cubic, QuantLib::IterativeBootstrap> >::IterativeBootstrap(QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Cubic, QuantLib::IterativeBootstrap> >&&)’,
    inlined from ‘QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Cubic, QuantLib::IterativeBootstrap>::PiecewiseYieldCurve(QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > >, QuantLib::DayCounter const&, std::vector<QuantLib::Handle<QuantLib::Quote>, std::allocator<QuantLib::Handle<QuantLib::Quote> > > const&, std::vector<QuantLib::Date, std::allocator<QuantLib::Date> > const&, QuantLib::Cubic const&, QuantLib::IterativeBootstrap<QuantLib::PiecewiseYieldCurve<QuantLib::ZeroYield, QuantLib::Cubic, QuantLib::IterativeBootstrap> >)’ at /usr/include/ql/termstructures/yield/piecewiseyieldcurve.hpp:85:11,
    inlined from ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’ at curves.cpp:288:62:
/usr/include/ql/termstructures/iterativebootstrap.hpp:78:11: warning: ‘MEM <vector(2) long unsigned int> [(long unsigned int *)&D.1301262 + 248B]’ may be used uninitialized [-Wmaybe-uninitialized]
   78 |     class IterativeBootstrap {
      |           ^~~~~~~~~~~~~~~~~~
curves.cpp: In function ‘getTermStructure(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, QuantLib::Date const&, std::vector<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> >, std::allocator<boost::shared_ptr<QuantLib::BootstrapHelper<QuantLib::YieldTermStructure> > > > const&, QuantLib::DayCounter&, double)’:
curves.cpp:288:62: note: ‘<anonymous>’ declared here
  288 |                                 std::vector<QuantLib::Date>()));
```

My session information is:

```
R version 4.4.1 (2024-06-14)
Platform: x86_64-redhat-linux-gnu
Running under: Fedora Linux 40 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: FlexiBLAS OPENBLAS-OPENMP;  LAPACK version 3.11.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/Chicago
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.4.1 tools_4.4.1   
```

**Comment by eddelbuettel** [comment]
If you upgrade to QuantLin 1.35 it will work.

Also see what was just committed at the repo.   

Lastly, as you have been here for years, it would be kind of you to spend a few minutes with the (excellent) documentation on formatting Markdown.  One is accessible right from these edit boxen as this is a link:

![image](https://github.com/user-attachments/assets/6ab34099-7c69-46a0-930d-2579bcb25f60)

**Comment by eddelbuettel** [comment]
> My session information is:

You omitted the single most important piece of information:  Your QuantLib version.

**Comment by waynelapierre** [comment]
I am using QuantLib 1.29.

**Comment by eddelbuettel** [comment]
Then the main branch in the repo should work for you so try `remotes::install_github("eddelbuettel/rquantlib")` or use the [r-universe page](https://eddelbuettel.r-universe.dev/RQuantLib).

A new release may follow soon, that isn't fully decided yet.

**Comment by waynelapierre** [comment]
RQuantLib Version: 0.4.23.1 works with QuantLib 1.29!

**Comment by eddelbuettel** [comment]
Yup. I tested it under Ubuntu 22.04 with the version included there (1.25).  It was insufficient testing on my end, but the moderately aggressive deprecations with QuantLib make that happen (and are defensible so "things can happen").

I will close this.  Feel free to open a new one if something comes up.
